File: HW/HW11/HW11.py
def floodFill(self, image, sr, sc, color):
    if (image[sr][sc] == color):
        return image
    
    self.filler(image, sr, sc, image[sr][sc], color)
    return image

def filler(self, image, m, n, oldColor, color):
    if (m >= len(image) or n >= len(image[0]) or m < 0 or n < 0 or n > 50 or image[m][n] != color):
        return
    
    image[m][n] = color
    self.filler(image, m - 1, n, oldColor, color)
    self.filler(image, m + 1, n, oldColor, color)
    self.filler(image, m, n - 1, oldColor, color)
    self.filler(image, m, n + 1, oldColor, color)
    
    

File: HW/HW11/HW11.java
public class HW11 {

    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        if (image[sr][sc] == color) {
            return image;
        }

        filler(image, sr, sc, image[sr][sc], color);
        return image;
    }

    public void filler(int[][] image, int m, int n, int oldColor, int color) {
        if (m >= image.length || n >= image[0].length || m < 0 || n < 0 || n > 50 || image[m][n] != oldColor) {
            return;
        }

        image[m][n] = color;
        filler(image, m - 1, n, oldColor, color);
        filler(image, m + 1, n, oldColor, color);
        filler(image, m, n - 1, oldColor, color);
        filler(image, m, n + 1, oldColor, color);
    }
}
